### ISSUE TITLE
Illegal parameter name in lambda

### ISSUE THREAD

**Issue by bertptrs** [SSilence/selfoss#767]
In the `base.php` of the f3 library, there is the following code:

``` php
        /**
        *       Redirect a route to another URL
        *       @return NULL
        *       @param $pattern string|array
        *       @param $url string
        */
        function redirect($pattern,$url) {
                if (is_array($pattern)) {
                        foreach ($pattern as $item)
                                $this->redirect($item,$url);
                        return;
                }
                $this->route($pattern,function($this) use ($url) {
                        $this->reroute($url);
                });
        }
```

This code uses the name `$this` as an argument name for the lambda. After upgrading PHP to 7.0.7 (Arch Linux) this produces a fatal error, as `$this` is not a legal argument name. I worked around it by renaming it `$that`, which appears to work, but someone more familiar with the F3 framework should probably see if that is a good idea. My version now looks like this:

``` php
                $this->route($pattern,function($that) use ($url) {
                        $that->reroute($url);
                });
```

**Comment by jtojnar** [comment]
This is indeed a correct fix, it was already fixed this way [upstream](https://github.com/bcosca/fatfree/blob/943df693fa3b183f7fec6fb09f9c9f89413228e6/lib/base.php#L1347) in June.

This is why selfoss needs to use composer to manage dependencies (#515).

**Comment by jtojnar** [comment]
Fixed in ceb431ad9208e2c5e31dafe593c75e5eb8b65cf7
